### PR TITLE
Fixing tests

### LINKED ISSUE #5: test cases
test cases in `arc/matrices/tests` can be greatly improved.

### PR DESCRIPTION
**Pull request by lautalaserna**
Let me know if its ok like this!

Closes #5

### CODE REVIEW

**Comment by Bchass** [comment]
Thanks, the new config was loaded in! Can you run these tests locally with `python -m pytest arc/matrices/tests` and investigate which ones are failing?

**Comment by lautalaserna** [comment]
Fixed the ones that not were working. Should be ok i think

**Comment by Bchass** [comment]
It's all good now 🚀  I'll review the test cases and leave comments if needed for further discussion

File: arc/matrices/tests/test_csr.py [hunk 1]
```diff
@@ -1,140 +1,231 @@
...
+        assert matrix_instance.matrix == expected_matrix, f"Expected {expected_matrix}, but got {matrix_instance.matrix}"
 
+        assert isinstance(matrix_instance, csr_matrix), "The object is not an instance of csr_matrix"
 
-def test_get_cols():
-    test_input_matrix = csr_matrix(3, 3).get_cols()
+    except Exception as e:
+        assert False, f"An error occurred: {e}"
+
+def test_data():
```
Comment (Bchass) [inline arc/matrices/tests/test_csr.py:33]: I originally removed this test case due to: 
```python3 
if len(data) != rows * cols:
                    raise ValueError("Length of data must match rows * cols.")
``` 

Wouldn't this still throw the ValueError?

File: arc/matrices/tests/test_csr.py [hunk 2]
```diff
@@ -1,140 +1,231 @@
...
+    try:
+        matrix_instance = csr_matrix(3, 3, data=[1, 2, 3, 4, 5, 6])
+        expected_matrix = [[1, 2, 3], [4, 5, 6], [0, 0, 0]]
+        
+        assert matrix_instance.matrix == expected_matrix, f"Expected {expected_matrix}, but got {matrix_instance.matrix}"
+        
+    except Exception as e:
+        assert False, f"An error occurred: {e}"
+
+def test_data_without_data():
```
Comment (Bchass) [inline arc/matrices/tests/test_csr.py:64]: this seems a little redundant to `test_default_constructor`

File: arc/matrices/tests/test_csr.py [hunk 3]
```diff
@@ -1,140 +1,231 @@
...
+    try:
+        matrix_instance = csr_matrix(0, 0)
 
-    assert test_input_matrix == expected_result
+        assert matrix_instance.matrix == [], "Expected an empty matrix"
 
+    except Exception as e:
+        assert False, f"An error occurred: {e}"
+
+def test_data_without_row_col():
```
Comment (Bchass) [inline arc/matrices/tests/test_csr.py:54]: same with `test_data`

File: arc/matrices/tests/test_csr.py (hunk 2 above)
Comment (lautalaserna) [reply arc/matrices/tests/test_csr.py:64]: Sure, did it base on what was already tested, but you are right

**Review by Bchass — CHANGES_REQUESTED** [review]
Please remove `test_data()`, `test_data_without_row_col()`, and `test_data_without_data()` and this will be good to go 🚀

**Comment by lautalaserna** [comment]
Done!

**Comment by Bchass** [comment]
I'll fix the one test that is failing for `nnz()`. Made some changes to it so I'm not surprised it's failing. Thanks again! @lautalaserna

**Pull request merged**